projects/habitat_ovmm/receptacles_data_collection.py

```diff
@@ -0,0 +1,267 @@
+                start_position, start_rotation, success = get_robot_spawns(target_positions=view_point_position[None], 
+                                                              rotation_perturbation_noise=0, 
+                                                              distance_threshold=0, 
+                                                              sim=sim, 
+                                                              num_spawn_attempts=100, 
+                                                              physics_stability_steps=100,
+                                                              orient_positions=recep_position[None],
+                                                            ) 
+                sim.robot.base_pos = start_position
+                sim.robot.base_rot = start_rotation
+                sim.maybe_update_robot()
+                obs = sim.get_sensor_observations()
+                plt.imsave(f'./data_out/scene_{scene_id}/ep_{episode.episode_id}/{recep}/img_{idx}.png', obs['robot_head_rgb'][:,:,:3])
+                idx += 1
+
+def gen_dataset_question(env):
+    # This is for iterating through all episodes once using only one env
+    count_dict = {}
+    for episode in env._dataset.episodes:
+        scene_id = extract_scene_id(episode.scene_id)
+        hash_str = f'ep_{episode.episode_id}_scene_{scene_id}'
+        count_dict[hash_str] = 0
+    
+    with open('./scene_info/recep_position.pickle', 'rb') as handle:
+        receptacle_positions = pickle.load(handle)
+
+    # Ideally, we can make it like an iterator to make it feel more intuitive
+    while True:
+        # Get a new episode
+        obs = env.reset()
+        episode = env.get_current_episode()
+        scene_id = extract_scene_id(episode.scene_id)
+
+        # Check if you have iterated through all episodes and if yes, break the loop
+        hash_str = f'ep_{episode.episode_id}_scene_{scene_id}'
+        if count_dict[hash_str] == 0:
+            count_dict[hash_str] += 1
+        elif count_dict[hash_str] == 1:
+            break
+        else:
+            raise NotImplementedError #Technically, this shouldn't happen
+
+        scene_receptacles = list(receptacle_positions[scene_id].keys())
+        scene_receptacles_id = dict()
+
+        recep_idx = 0
+        for recep in scene_receptacles:
+            scene_dir = f'./data_out/scene_{scene_id}/ep_{episode.episode_id}/{recep}/'
+            scene_receptacles_id[recep] = []
+
+            for path in os.listdir(scene_dir):
+                if os.path.isfile(os.path.join(scene_dir, path)):
+                    scene_receptacles_id[recep].append(recep_idx)
+                    recep_idx += 1
+        
+        for (q_idx, recep) in enumerate(all_receptacles):
+            question_str = f"We show images from different locations inside a home. Which location(s) contain {recep.replace('_', ' ')}? You can choose multiple options\n"
```

Comment:
Would be good to generate some questions with objects too? These are fairly large + well positioned

Reply:
Should I include questions of type "Does this image contain <insert object name>?" Feel free to suggest other types of questions as well

Reply:
I think this is a great idea